add-sailthru-paths.js

```diff
@@ -47,7 +47,7 @@ function isNotSailthruPattern(pattern) {
  */
 function encodePattern(pattern) {
   const path = pattern.match(/^NOT (.+)$/)[1]
-  const encoded = encodePath(path.slice(0, -(path.length % 3)))
+  const encoded = encodePath(path.slice(0, path.length - (path.length % 3)))
```

Comment:
This is the fix. [The docs for `slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) describe how it works when the argument is negative – it's a shorthand for what I've written out here. The bug was that, if `path.length` is _exactly_ a multiple of 3, then `-(path.length % 3)` evaluates to `-0`, and `slice()` returns `""`.

Reply:
that's a hell of a bugfix @ashfurrow 👍